app/src/full/kotlin/io/homeassistant/companion/android/settings/wear/SettingsWearViewModel.kt

```diff
@@ -411,4 +422,36 @@ class SettingsWearViewModel @Inject constructor(private val serverManager: Serve
             _resultSnackbar.emit(app.getString(commonR.string.failed_watch_connection))
         }
     }
+
+    fun updateWearNodesWithApp(nodes: Set<Node>?) {
+        _wearNodesWithApp.value = nodes
+    }
+
+    suspend fun findWearDevicesWithApp(capabilityClient: CapabilityClient?) {
+        try {
+            val capabilityInfo = capabilityClient
+                ?.getCapability(CAPABILITY_WEAR_APP, CapabilityClient.FILTER_ALL)
+                ?.await()
```

Comment:
It's nice to move the logic in the VM, but you did it partially so `CAPABILITY_WEAR_APP` is duplicated for instance.